raphtory/src/serialise/serialise.rs

```diff
@@ -1456,4 +1475,148 @@ mod proto_test {
             Prop::from_arr::<UInt8Type>(vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
         ));
     }
+
+    #[cfg(feature = "search")]
+    mod test_index_io {
+        use crate::{
+            core::{utils::errors::GraphError, Prop},
+            db::{
+                api::{
+                    mutation::internal::{InternalAdditionOps, InternalPropertyAdditionOps},
+                    view::{internal::InternalStorageOps, StaticGraphViewOps},
+                },
+                graph::views::filter::{AsNodeFilter, NodeFilter, NodeFilterBuilderOps},
+            },
+            prelude::{
+                AdditionOps, Graph, GraphViewOps, NodeViewOps, PropertyAdditionOps,
+                SearchableGraphOps, StableDecode, StableEncode,
+            },
+            serialise::GraphFolder,
+        };
+        use raphtory_api::core::storage::arc_str::ArcStr;
+
+        fn init_graph<G>(graph: G) -> G
+        where
+            G: StaticGraphViewOps
+                + AdditionOps
+                + InternalAdditionOps
+                + InternalPropertyAdditionOps
+                + PropertyAdditionOps,
+        {
+            graph
+                .add_node(
+                    1,
+                    "Alice",
+                    vec![("p1", Prop::U64(2u64))],
+                    Some("fire_nation"),
+                )
+                .unwrap();
+            graph
+        }
+
+        fn assert_search_results<T: AsNodeFilter + Clone>(
+            graph: &Graph,
+            filter: &T,
+            expected: Vec<&str>,
+        ) {
+            let res = graph
+                .search_nodes(filter.clone(), 2, 0)
+                .unwrap()
+                .into_iter()
+                .map(|n| n.name())
+                .collect::<Vec<_>>();
+            assert_eq!(res, expected);
+        }
+
```

Comment:
make prop tests for the different cases (folder and zip, PersistentGraph, Graph, index, no index)